### PR TITLE
ARROW-12393: [JS] [WIP] Drop closure compiler

### PR DESCRIPTION
**Pull request by domoritz**
Before

```
693k -I Arrow.es5.min.js.map
 55k -I Arrow.externs.js
252k -I Arrow.js
```

After

```
1.2M -I Arrow.es5.min.js.map
310k -I Arrow.js
```

### CODE REVIEW

File: js/gulpfile.js [hunk 1]
```diff
@@ -46,13 +44,10 @@ for (const [target, format] of combinations([`all`], [`all`])) {
...
     const umd = taskName(target, `umd`);
-    const cls = taskName(UMDSourceTargets[target], `cls`);
+    const esm = taskName(target, `esm`);
     gulp.task(`build:${umd}`, gulp.series(
-        `build:${cls}`,
-        `clean:${umd}`, `compile:${umd}`, `package:${umd}`,
-        function remove_closure_tmp_files() {
-            return del(targetDir(target, `cls`))
-        }
+        `build:${esm}`,
```
Comment (domoritz) [inline js/gulpfile.js:49]: @trxcllnt I don't know much about gulp but this seems to run the same job twice when I run `yarn build -t es5`. How can I avoid that?

**Comment by domoritz** [comment]
Maybe it's worth trying to run closure over the es2015 sources as well (instead of removing closure) if the results are that much better 😆.

**Comment by alippai** [comment]
@domoritz closure compiler is discontinued, right? I can help looking into alternatives if you need any help with this issue.

**Comment by domoritz** [comment]
Thanks for offering to help. No, I don't think it's discontinued but it's a lot of work to keep it running for potentially small benefits so we are considering to remove it. We already use terser for all non es5 builds.

**Comment by alippai** [comment]
https://github.com/google/closure-compiler-npm/blob/master/packages/google-closure-compiler-js/readme.md, you are right, it looks like only the JS version is being discontinued

**Comment by domoritz** [comment]
Ah yes, I switched away from that last month when we updated to the latest closure compiler version.

**Comment by trxcllnt** [comment]
@domoritz it seems like not using Closure Compiler yields a bundle that's 58kb larger. Am I reading that correctly? What's the impact after compression? A larger bundle seems like a regression here.

**Comment by domoritz** [comment]
I haven't had the cycles to actually check. The bundles may not be correct yet so I wouldn't count on these numbers yet. 

But if they are right, I wonder whether we should use closure for non es5 as well.

**Comment by domoritz** [comment]
Will redo and compare with updated terser as well.

**Pull request closed without merging**